docs/learn-js/Promises.md

```diff
@@ -0,0 +1,51 @@
+# Promises
+
+Dealing with promises and async libs in CT.JS
+can be confusing.. To simplify the idea of async code,
+place wrapper function in the global scope 
+** Project - Custom Scripts **
+
+## Example problem
+
+You have a code library you wish to import into CT.JS
+that uses an async function called someFunction(someVars).
+You cannot directly call this function within an object
+because you need to use an **await** keyword.
+
+## Solution
+
+* Create Wrapper in ** Custom Scrips **
+	```
+	async function myFunction(someVars,passCallback,failCallback,alwaysCallback){
+	try{
+		const reply=await someFunction(someVars);
+		passCallback(reply);
+	}
+	catch(error){
+		failCallback("Error: "+error.message);
+	}
+	finally(){
+		alwaysCallback(someVars);
+	}
+	```
+* Create callback functions in object's **Creation** event
+	```
+	this.myPass=function(myVar){
+		console.log(myVar);
+	}
+	this.myFail=function(myError){
+		console.log(myError);
+	}
+	this.myAlway=function(myVar){
+		console.log(myVar);
+	}
+	```
+* Now you can call your Wrapper from your copy
+	`myFunction("SomeCoolData",this.myPass,this.myFail,this.myAlway);`

```

Comment:
This is anything but simplification, and it has nothing to do with ct.js as well. Ct.js uses promises with ct.tween, ct.transition, and ct.u.wait — would be cool if these were covered instead.

Reply:
> You cannot directly call this function within an object
because you need to use an **await** keyword.

You don't. You can use `.then` and `.catch`.